pkg/controller/reconciler.go

```diff
@@ -68,19 +68,27 @@ func (o *operationAnnotationWrapper) Reconcile(request reconcile.Request) (recon
 		return reconcile.Result{}, err
 	}
 
+	if err := o.removeOperationAnnotation(obj, v1beta1constants.GardenerOperation, v1beta1constants.GardenerOperationReconcile, v1beta1constants.GardenerOperationRestore); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return o.Reconciler.Reconcile(request)
+}
+
+func (o *operationAnnotationWrapper) removeOperationAnnotation(obj runtime.Object, key string, values ...string) error {
```

Comment:
If you keep this functionality in #528, I would design the function more generally:

```
func RemoveAnnotationContainingValues(ctx context.Context, client client.Client, obj runtime.Object, key string, values ...string) error
```